tests/python/test_field.py

```diff
@@ -99,6 +99,37 @@ def test_scalr_field_from_numpy_with_mismatch_shape(dtype, shape):
         x.from_numpy(arr)
 
 
+@pytest.mark.parametrize('dtype', data_types)
+@pytest.mark.parametrize('shape', field_shapes)
+@ti.test(arch=ti.get_host_arch_list())
+def test_field_copy_from(dtype, shape):
+    x = ti.field(dtype, shape)
+    other = ti.field(dtype, shape)
```

Comment:
can we also initialize `other` differently and assert on the copied `x`'s value?

Reply:
Since copying between different shapes is allowed, how do I assert the copied value? Comparing the overlap? 

And `field.copy_from` calls `taichi.lang.meta.tensor_to_tensor` to perform the copy, maybe it's better to do that test in `meta`'s test file.

Reply:
oh good point! Sorry I just noticed that we allow copying from a different shape hmmm...yea in that case it's even nicer to have tests covering this. You can simply assert what's the expected `x` value directly. Thanks a lot!

Reply:
Hi Ail,

Can you confirm if copying from a different shape is allowed in ```copy_from```? 
The original code only assert the length of the shape is equal, but the comment says 
> The shape of the other field needs to be the same as `self`.

And if it's allowed, I am confused about the intended usage of ```copy_from```.

```
>> x = ti.field(ti.f32, (5,5))
>> y = ti.field(ti.f64, (4,3))
>> y.fill(1)
>> x.copy_from(y); x.to_numpy()
array([[1., 1., 1., 0., 1.],
       [1., 1., 1., 0., 1.],
       [1., 1., 1., 0., 1.],
       [1., 1., 1., 0., 1.],
       [1., 1., 1., 0., 1.]], dtype=float32)
```
It's not simply padded with 0.

Reply:
Hi @ljjsalt I'd prefer to ban this behavior as long as it passes tests ;) (in other words, nobody depends on this behavior).
I do agree that it's confusing to allow copying from a different shape.